internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -249,35 +255,10 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	// DependsOn with the bundle's DependsOn (pure function) and replacing
 	// the labels with the bundle's labels
 	for _, target := range matchedTargets {
-		if target.Deployment == nil {
-			continue
-		}
-		if target.Deployment.Namespace == "" {
-			logger.V(1).Info("Skipping bundledeployment with empty namespace, waiting for agentmanagement to set cluster.status.namespace", "bundledeployment", target.Deployment)
-			continue
-		}
-
-		// NOTE we don't use the existing BundleDeployment, we discard annotations, status, etc
-		// copy labels from Bundle as they might have changed
-		bd := target.BundleDeployment()
-
-		// No need to check the deletion timestamp here before adding a finalizer, since the bundle has just
-		// been created.
-		controllerutil.AddFinalizer(bd, bundleDeploymentFinalizer)
-
-		bd.Spec.OCIContents = contentsInOCI
-
-		updated := bd.DeepCopy()
-		op, err := controllerutil.CreateOrUpdate(ctx, r.Client, bd, func() error {
-			bd.Spec = updated.Spec
-			bd.Labels = updated.GetLabels()
-			return nil
-		})
-		if err != nil && !apierrors.IsAlreadyExists(err) {
-			logger.Error(err, "Reconcile failed to create or update bundledeployment", "bundledeployment", bd, "operation", op)
+		bd, err := r.createBundle(ctx, logger, target, contentsInOCI)
```

Comment:
Since createBundle can return a nil bd, do we still need the `continue`. Apparently it's only passed to createDeploymentOCISecret, so if that can handle nil, we probably don't need `continue`..?

Reply:
What I do see now from reading your comment and thinking about it is that since `createBundle` will return a `nil` bundle with a `nil` error in cases where we would previously `continue`, it would make sense to add `bd != nil` to the condition on L263, to avoid calling `createDeploymentOCISecret` in such cases.
While `createDeploymentOCISecret` can perhaps handle `nil` bundle deployments, it may error when getting a secret or setting ownership of a secret on a `nil` bundle, so it would make sense to avoid that.

How close is this to what you meant?